### DISCUSSION TITLE
Linter: flag to avoid emitting warnings

### DISCUSSION THREAD

**Discussion by Vivalldi** [biomejs/biome#550] [Ideas]
Some teams prefer certain rules to warn rather than error. These warnings are extremely useful in the editor to prompt developers to double check their work. Warnings in the `biome lint` output can become overbearing at times when attempting to find diagnostics that are set to `error`. An optional flag to hide warnings from the output would help locate errors more quickly.

- `--hide-warnings`
- `--skip-warnings` (might cause confusion with `--skip-errors`)
- `--omit-warnings`
- `--errors-only`

**Comment by ematipico** [comment]
I like the proposal, I think it makes sense. 

About the naming, since we added `--log-level`, we could stick to that naming convention and use `--diagnostic-level`, and the order will be as follows (from lowest to highest):
- Fatal
- Panic
- Info
- Warning
- Error

This means that if you pass `--diagnostic-level=info` you will see only info, warnings and errors. If you pass `--diagnostic-level=error` you will see only errors. What do you think?

**Reply by Vivalldi** [reply]
That name makes perfect sense! Not sure I understand why `fatal` & `panic` are lowest instead of highest.

**Reply by ematipico** [reply]
Yeah you're right, they should be the hightest

**Comment by mshahzebraza** [comment]
Any updates on this?

**Reply by siketyan** [reply]
We have a CLI flag `--diagnostic-level` now: https://biomejs.dev/reference/cli/#biome-check
